src/main/java/sonarquberepair/Main.java

```diff
@@ -1,66 +1,106 @@
 
-	public static void repair(String pathToFile, String projectKey, int ruleKey, PrettyPrintingStrategy prettyPrintingStrategy) throws Exception {
-		Launcher launcher = new Launcher();
-		launcher.addInputResource(pathToFile);
-		launcher.getEnvironment().setAutoImports(true);
-		if (prettyPrintingStrategy == PrettyPrintingStrategy.SNIPER) {
-			launcher.getEnvironment().setPrettyPrinterCreator(() -> {
-						SniperJavaPrettyPrinter sniper = new SniperJavaPrettyPrinter(launcher.getEnvironment());
-						sniper.setIgnoreImplicit(false);
-						return sniper;
-					}
-			);
-			launcher.getEnvironment().setCommentEnabled(true);
-			launcher.getEnvironment().useTabulations(true);
-			launcher.getEnvironment().setTabulationSize(4);
-		}
-
-		Class<?> processor = Processors.getProcessor(ruleKey);
-		Constructor<?> cons;
-		Object object;
-		try {
-			cons = processor.getConstructor(String.class);
-			object = cons.newInstance(projectKey);
-		} catch (NoSuchMethodException e) {
-			cons = processor.getConstructor();
-			object = cons.newInstance();
-		}
-		launcher.addProcessor((Processor) object);
-		launcher.run();
+		opt = new FlaggedOption("workspace");
+		opt.setLongFlag("workspace");
+		opt.setStringParser(JSAP.STRING_PARSER);
+		opt.setDefault("./sonar-workspace");
+		opt.setHelp("Workspace of SonarQubeRepair");
+		jsap.registerParameter(opt);
+
+		opt = new FlaggedOption("gitRepoPath");
+		opt.setLongFlag("gitRepoPath");
+		opt.setStringParser(JSAP.STRING_PARSER);
+		opt.setHelp("Root Path of the Git repo directory");
+		jsap.registerParameter(opt);
+
+		return jsap;
+	}
+
+	public void initConfig(JSAP jsap,String[] args) {
+		JSAPResult jsapRes = jsap.parse(args);
+
+		this.getConfig().addRuleKeys(jsapRes.getInt("ruleKeys"));
+		this.getConfig().setProjectKey(jsapRes.getString("projectKey"));
+		this.getConfig().setOriginalFilesPath(jsapRes.getString("originalFilesPath"));
+		this.getConfig().setPrettyPrintingStrategy(PrettyPrintingStrategy.valueOf(jsapRes.getString("prettyPrintingStrategy")));
+		this.getConfig().setFileOutputStrategy(FileOutputStrategy.valueOf(jsapRes.getString("fileOutputStrategy")));
+		this.getConfig().setWorkspace(jsapRes.getString("workspace"));
+		this.getConfig().setGitRepoPath(jsapRes.getString("gitRepoPath"));
+	}
+
+
+	public DefaultRepair getRepairProcess() {
+		DefaultRepair defaultRepair = new DefaultRepair(this.config);
+		return defaultRepair;
 	}
 
 	/**
-	 * @param args string array.
-	 *             Give one or two arguments.
-	 *             The first argument is the Sonar rule key.
-	 *             The second argument is the project key for the Sonar analysis of source files.
+	 * @param args string array. Give either 0, 1 or 2 arguments. first argument is sonarqube rule-number which you can get from https://rules.sonarsource.com/java/type/Bug
+	 *             second argument is the projectKey for the sonarqube analysis of source files. for  example "fr.inria.gforge.spoon:spoon-core"
```

Comment:
Henry, you should undo this change in the comment. I believe you didn't do this change on purpose, it happened in some reset you did or something like that.

Reply:
removed again :) .